libcontainer/integration/exec_test.go

```diff
@@ -932,3 +932,52 @@ func TestOomScoreAdj(t *testing.T) {
 		t.Fatalf("Expected oom_score_adj %d; got %q", config.OomScoreAdj, outputOomScoreAdj)
 	}
 }
+
+func TestPrestartHook(t *testing.T) {
+	if testing.Short() {
+		return
+	}
+	root, err := newTestRoot()
+	ok(t, err)
+	defer os.RemoveAll(root)
+
+	rootfs, err := newRootfs()
+	ok(t, err)
+	defer remove(rootfs)
+
+	config := newTemplateConfig(rootfs)
+	config.Hooks = &configs.Hooks{
+		Prestart: []configs.Hook{
+			configs.NewFunctionHook(func(s *configs.HookState) error {
+				f, err := os.Create(filepath.Join(rootfs, "test"))
```

Comment:
dont think we should reach out to rootfs from here. We should pass it as hook state as @mrunalp mentioned before.